### PR TITLE
Remove `getResourceUri` and `getSasToken` util functions, add them to `ITransferSrcOrDstTreeItem` (formerly `IDownloadableTreeItem`) instead

### PR DESCRIPTION
**Pull request by MRayermannMSFT**

### CODE REVIEW

File: src/commands/downloadFiles/GetAzCopyDownloadsStep.ts [hunk 1]
```diff
@@ -50,7 +47,7 @@ export class GetAzCopyDownloadsStep extends AzureWizardExecuteStep<IDownloadWiza
                     remoteFilePath,
                     localFilePath: join(destinationFolder, treeItem.blobName),
                     isDirectory: false,
-                    resourceUri,
+                    resourceUri: treeItem.resourceUri,
```
Comment (nturinski) [inline src/commands/downloadFiles/GetAzCopyDownloadsStep.ts:50]: nit: Could create a `const resourceUri = treeItem.resourceUri` in the for loop to keep the objects being pushed cleaner.

File: src/tree/blob/BlobDirectoryTreeItem.ts [hunk 2]
```diff
@@ -63,6 +64,21 @@ export class BlobDirectoryTreeItem extends AzExtParentTreeItem implements ICopyU
         return new vscode.ThemeIcon('folder');
     }
 
+    public get resourceUri(): string {
```
Comment (nturinski) [inline src/tree/blob/BlobDirectoryTreeItem.ts:67]: As far as I can tell, all of the blobTreeItems share the same `getters` for `resourceUri` and `transferSasToken`. Would you be open to creating a base class that implements it that could just be extended?

File: src/tree/fileShare/DirectoryTreeItem.ts [hunk 3]
```diff
@@ -51,6 +52,21 @@ export class DirectoryTreeItem extends AzExtParentTreeItem implements ICopyUrl,
         return path.posix.join(this.parentPath, this.directoryName);
     }
 
+    public get resourceUri(): string {
```
Comment (nturinski) [inline src/tree/fileShare/DirectoryTreeItem.ts:55]: Similar comment, but for fileshares.

**Review by nturinski — DISMISSED** [review]
I'd be okay if you didn't want to abstract the commands out since this is already a clean, vast improvement.

File: src/tree/fileShare/DirectoryTreeItem.ts (hunk 3 above)
Comment (MRayermannMSFT) [reply src/tree/fileShare/DirectoryTreeItem.ts:55]: Will leave for someone who can take a look across all storage tree item classes.

File: src/tree/blob/BlobDirectoryTreeItem.ts (hunk 2 above)
Comment (MRayermannMSFT) [reply src/tree/blob/BlobDirectoryTreeItem.ts:67]: Will leave for someone who can take a look across all storage tree item classes.

**Review by nturinski — APPROVED** [review]

**Pull request merged**